### PR TITLE
Ipm v3

### PR DESCRIPTION
**Pull request by edersondisouza**
Some IPM samples for Zephyr. Depends on https://github.com/solettaproject/soletta/pull/2006

### CODE REVIEW

File: ipm/arc/src/main.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+static bool
+timeout_send_cb(void *data)
+{
+    char *str = strdup(samples[count % 4]);
+    struct sol_blob *message = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL,
+        str, strlen(str) + 1);
+
+    LOG("ARC sending %s - %p\n", message->mem, message->mem);
+    sol_ipm_send(MESSAGE_ID, message);
```
Comment (glima) [inline ipm/arc/src/main.c:47]: I'd check for return here

File: ipm/x86/src/main.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+static bool
+timeout_send_cb(void *data)
+{
+    struct sol_blob *message = sol_blob_new(SOL_BLOB_TYPE_NO_FREE, NULL,
+        samples[count % 4], strlen(samples[count % 4]));
+
+    SOL_WRN("x86 sending %p - %s", message->mem, message->mem);
+    sol_ipm_send(MESSAGE_ID, message);
```
Comment (glima) [inline ipm/x86/src/main.c:53]: Ditto

File: ipm/README [hunk 3]
```diff
@@ -0,0 +1,22 @@
+This samples shows a simple communication between two cores
+on Zephyr. It can be tested on an Arduino 101, for instance.
+This board have two cores: one ARC and one Quark (x86).
+Directories `x86/` and `arc/` have program that runs on x86 and ARC
+core, respectively.
```
Comment (glima) [inline ipm/README:5]: Fine, but I'd state what it really does: sends strings from both sides indefinitely.

File: ipm-flow/x86/src/main.fbp [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ipm_empty_writer(ipm/empty-writer:id=8)
+timer_empty(timer:interval=4000)
+timer_empty OUT -> IN _(converter/empty-to-boolean:output_value=false) OUT -> ENABLED timer_empty
+
+timer_empty OUT -> IN ipm_empty_writer
+
+ipm_empty_writer CONSUMED -> IN _(console:prefix="x86 empty write consumed!")
+
+ipm_empty_reader(ipm/empty-reader:id=8)
+ipm_empty_reader OUT -> IN empty_reader_x86(console:prefix="x86 got: ")
```
Comment (glima) [inline ipm-flow/x86/src/main.fbp:106]: OK. Don't we get a more real-life sample, like reading ADC samples from ARC and sending them to LKM?

Comment (edersondisouza) [inline ipm-flow/x86/src/main.fbp:106]: It's possible to do: here I kinda converted a test (once it tests all IPM node types) to sample. Nevertheless, I think this one has its value: it's really easy to test, do not demand any wiring.

File: ipm/README (hunk 3 above)
Comment (edersondisouza) [inline ipm/README:5]: I state that on last paragraph... but I may highlight it.

**Comment by edersondisouza** [comment]
v4 on #20

**Pull request closed without merging**